internal/provider/resource_switch_port_settings.go

```diff
@@ -0,0 +1,1304 @@
+				}(),
+				RemoteASN: func() types.Int64 {
+					if bgpPeer.RemoteAsn == nil {
+						return types.Int64Null()
+					}
+					return types.Int64Value(int64(*bgpPeer.RemoteAsn))
+				}(),
+				VlanID: func() types.Int32 {
+					if bgpPeer.VlanId == nil {
+						return types.Int32Null()
+					}
+					return types.Int32Value(int32(*bgpPeer.VlanId))
+				}(),
+			}
+
+			bgpPeerModel.AllowedExport = &switchPortSettingsBGPPeerPeerAllowedExportModel{
+				Type: types.StringValue(string(bgpPeer.AllowedExport.Type)),
+				Value: func() []types.String {
+					res := make([]types.String, 0)
+					for _, elem := range bgpPeer.AllowedExport.Value {
+						res = append(res, types.StringValue(elem.(string)))
+					}
+					return res
+				}(),
+			}
+
+			bgpPeerModel.AllowedImport = &switchPortSettingsBGPPeerPeerAllowedImportModel{
+				Type: types.StringValue(string(bgpPeer.AllowedImport.Type)),
+				Value: func() []types.String {
+					res := make([]types.String, 0)
+					for _, elem := range bgpPeer.AllowedImport.Value {
+						res = append(res, types.StringValue(elem.(string)))
+					}
+					return res
+				}(),
+			}
+
+			bgpPeerModel.Communities = func() []types.Int64 {
+				communities := make([]types.Int64, 0)
+				for _, communityStr := range bgpPeer.Communities {
+					community, err := strconv.ParseInt(communityStr, 10, 64)
+					if err != nil {
+						diags.AddError(
+							"Error parsing community element",
+							fmt.Sprintf("Could not parse %s as int64: %v", communityStr, err),
+						)
+					}
+					communities = append(communities, types.Int64Value(community))
+				}
+				return communities
+			}()
+
+			linkToBGPPeer[link] = append(linkToBGPPeer[link], bgpPeerModel)
+		}
+
+		bgpPeersModels := make([]switchPortSettingsBGPPeerModel, 0)
+		for linkName, bgpPeers := range linkToBGPPeer {
+			bgpPeerModel := switchPortSettingsBGPPeerModel{
+				Peers:    bgpPeers,
+				LinkName: types.StringValue(linkName),
+			}
+			bgpPeersModels = append(bgpPeersModels, bgpPeerModel)
+		}
+
+		model.BGPPeers = bgpPeersModels
+	}
+
+	//
+	// Groups
+	//
+	if len(settings.Groups) > 0 {
+		groupModels := make([]types.String, 0)
+		for _, group := range settings.Groups {
+			groupModel := types.StringValue(group.PortSettingsGroupId)
+			groupModels = append(groupModels, groupModel)
+		}
+		model.Groups = groupModels
+	}
+
+	//
+	// Interfaces
+	//
+	if len(settings.Interfaces) > 0 {
+		interfaceModels := make([]switchPortSettingsInterfaceModel, 0)
+		for _, iface := range settings.Interfaces {
+			interfaceModel := switchPortSettingsInterfaceModel{
+				Kind: &switchPortSettingsInterfaceKindModel{
+					Type: types.StringValue(string(iface.Kind)),
+				},
+				LinkName:  types.StringValue(string(iface.InterfaceName)),
+				V6Enabled: types.BoolPointerValue(iface.V6Enabled),
+			}
+			interfaceModels = append(interfaceModels, interfaceModel)
+		}
+		model.Interfaces = interfaceModels
+	}
+
+	//
+	// Links
+	//
+	if len(settings.Links) > 0 {
```

Comment:
Same

Reply:
Answered in https://github.com/oxidecomputer/terraform-provider-oxide/pull/426#discussion_r2212336366. Going to consolidate discussion there.